tools/export/makefile/__init__.py

```diff
@@ -139,12 +148,14 @@ def generate(self):
         # Add the virtual path the the include option in the ASM flags
         new_asm_flags = []
         for flag in ctx['asm_flags']:
-            if flag.startswith('-I'):
-                new_asm_flags.append("-I{}/{}".format(ctx['vpath'][0], flag[2:]))
-            elif flag.startswith('--preinclude='):
-                new_asm_flags.append("--preinclude={}/{}".format(ctx['vpath'][0], flag[13:]))
+            if flag.startswith('--cpreproc_opts='):
+                sub_flags = flag.split(',')
+                new_sub_flags = []
+                for sub_flag in sub_flags:
+                    new_sub_flags.append(_fix_include_asm_flag(sub_flag, ctx))
+                new_asm_flags.append(','.join(new_sub_flags))
```

Comment:
one question - how does this fixes the issue - why paths for asm preprocessed files were not correct? I can see magic above based on -I or preinclude - shall we add some details to the commit msg?